lib/child-writer/index.js

```diff
@@ -113,25 +113,36 @@ exports.write = (image, drive, options) => {
     emitter.emit('error', error);
   };
 
-  ipc.server.on('error', emitError);
-  ipc.server.on('message', (data) => {
-    let message = null;
-
+  /**
+   * @summary Bridge robot message to the child writer caller
+   * @function
+   * @private
+   *
+   * @param {String} message - robot message
+   *
+   * @example
+   * bridgeRobotMessage(robot.buildErrorMessage(new Error('foo')));
+   */
+  const bridgeRobotMessage = (message) => {
     try {
-      message = robot.parseMessage(data);
-    } catch (error) {
-      return emitError(error);
-    }
+      const parsedMessage = robot.parseMessage(message);
+
+      // The error object is decomposed by the CLI for serialisation
+      // purposes. We compose it back to an `Error` here in order
+      // to provide better encapsulation.
+      if (robot.getCommand(parsedMessage) === 'error') {
+        emitError(robot.recomposeErrorMessage(parsedMessage));
+      } else {
+        emitter.emit(robot.getCommand(parsedMessage), robot.getData(parsedMessage));
```

Comment:
Fair enough. Perhaps I'd be happier if there was an inline code-comment mentioning that `robot.getCommand` and `robot.getData` are 'lightweight' accessor methods?

The reason I was concerned when I first saw
```
emitter.emit(robot.getCommand(parsedMessage), robot.getData(parsedMessage));
```
was because I was worried that `robot.getCommand` and `robot.getData` were possibly 'heavy' methods, which might be (internally) repeating the same code-actions to pull the required information out of `parsedMessage`. IYSWIM?